### PR TITLE
More trolling

### PR DESCRIPTION
**Pull request by mat-1**
Fixes stuff clippy complains about and add fake lag when sending packets to the server (random between 0-reconnects*1000ms)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -79,51 +71,121 @@ async fn transfer(
...
+        .read()
+        .await
+        .database
+        .data
+        .iter()
+        .find(|c| c.ip == split[0])
+        .map(|c| c.logins.len())
+        .unwrap_or(0);
+
+    println!("reconnects: {}", reconnects);
```
Comment (cleonyc) [inline src/main.rs]: Remove this logging or make it more useful

File: src/main.rs [hunk 2]
```diff
@@ -79,51 +71,121 @@ async fn transfer(
...
+        // first join, 10-30 minutes
+        (rand::random::<f64>() * 60.0 * 20.0) as u64 + 10 * 60
+    } else {
+        // they've joined before, 10-130s
+        (rand::random::<f64>() * 60.0 * 2.0) as u64 + 10
+    };
+
+    // ping is random between 0 and reconnects*1000ms
+    let simulated_ping: u64 = (rand::random::<f32>() * (reconnects as f32) * 1000.0) as u64;
+    println!("simulated_ping: {}ms", simulated_ping);
```
Comment (cleonyc) [inline src/main.rs]: Same thing goes for this.

File: src/main.rs [hunk 3]
```diff
@@ -79,51 +71,121 @@ async fn transfer(
...
+            match message {
+                Ok(bytes) => {
+                    if write_inbound.write_all(&bytes).await.is_err() {
+                        break;
+                    }
+                }
+                Err(_) => break,
+            }
+        }
+
```
Comment (cleonyc) [inline src/main.rs:175]: is there a reason to do this over io::copy, as imo io::copy is more descriptive, plus it uses less ram.

**Review by cleonyc — CHANGES_REQUESTED** [review]
looks very good

File: src/main.rs (hunk 1 above)
Comment (mat-1) [reply src/main.rs]: Oops yeah forgot to remove these while I was testing

File: src/main.rs (hunk 3 above)
Comment (mat-1) [reply src/main.rs:175]: I'm pretty sure it uses the same amount of ram, I just made it like that so it's easier to add fake lag for server->client packets later.

**Pull request merged**